Backport to 1.17.8 of patch submitted & accepted upstream in:
https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/599

From 78f7dbcba8c068d5383de9a8aec82d7bb3cfdda3 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Subject: [PATCH] Fix cairo-perf-print build on Solaris
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Adds include of <termios.h> if system has it, in order to fix:

../perf/cairo-perf-print.c: In function ‘report_print’:
../perf/cairo-perf-print.c:62:28: error: storage size of ‘ws’ isn’t known
   62 |             struct winsize ws;
      |                            ^~
../perf/cairo-perf-print.c:64:26: error: ‘TIOCGWINSZ’ undeclared
 (first use in this function)
   64 |             if(ioctl(fd, TIOCGWINSZ, &ws) == 0 ) {
      |                          ^~~~~~~~~~

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 meson.build             | 1 +
 perf/cairo-perf-print.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index e45e441f3..162b2927c 100644
--- a/meson.build
+++ b/meson.build
@@ -138,5 +138,6 @@ check_headers = [
   ['sys/ioctl.h'],
   ['intsafe.h'],
+  ['termios.h'],
 ]
 
 check_types = [
diff --git a/perf/cairo-perf-print.c b/perf/cairo-perf-print.c
index ab1440f55..e9c493fa3 100644
--- a/perf/cairo-perf-print.c
+++ b/perf/cairo-perf-print.c
@@ -43,6 +43,9 @@
 #if USE_TERMINAL_SIZE
 #include <unistd.h>
 #include <sys/ioctl.h>
+#if HAVE_TERMIOS_H
+#include <termios.h>
+#endif
 #endif
 
 static void
-- 
2.45.2

